File: data_structures/find_missing_number.js
// find missing number

const findMissing = arr => {
    const total = (arr.length + 1) * (arr.length + 2) / 2
    const sum = arr.reduce((t, i) => t += i)
    console.log(total, sum)
    return total - sum
}

const test1 = [1, 2, 3, 5]
console.log(findMissing(test1))

File: data-structures/find_missing_number.cpp
#include <iostream>
#include <vector>

using namespace std;

int findMissing (vector<int> arr) {
    int total = (arr.size() + 1) * (arr.size() + 2) / 2, sum = 0;
    for (int i : arr) sum += i;
    return total - sum;
}

int main () {
    vector<int> test1{ 1, 2, 3, 4, 5, 6, 7, 8, 10 };
    cout << findMissing(test1) << endl;
}